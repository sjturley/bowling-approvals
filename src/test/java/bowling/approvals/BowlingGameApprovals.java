/*
 * This Java source file was generated by the Gradle 'init' task.
 */
package bowling.approvals;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.util.Arrays;

import static org.junit.jupiter.api.Assertions.*;

class BowlingGameApprovals {
    @Test
    public void scoring() {
        Integer[][] games = {{}
                , {5}
                , {0}
                , {1, 2}
                , {3, 4, 5, 1, 7, 0}
                , {5, 5}
                , {5, 5, 5}
                , {5, 5, 3, 7, 4}
                , {10}
        };
        Approvals.verifyAll("", games, array -> {
            BowlingGame bowlingGame = new BowlingGame();
            for (int roll : array) {
                bowlingGame.roll(roll);
            }
            return Arrays.toString(array) + "\n" + bowlingGame.score() + "\n";
        });
    }
}
