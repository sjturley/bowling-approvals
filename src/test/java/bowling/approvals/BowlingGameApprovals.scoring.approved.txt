[]
 ____________________________________________________________
|__1__|__2__|__3__|__4__|__5__|__6__|__7__|__8__|__9__|__10__|
|   |_|   |_|   |_|   |_|   |_|   |_|   |_|   |_|   |_|  |_|_|
|     |     |     |     |     |     |     |     |     |      |
|_____|_____|_____|_____|_____|_____|_____|_____|_____|______|

[5]
 ____________________________________________________________
|__1__|__2__|__3__|__4__|__5__|__6__|__7__|__8__|__9__|__10__|
|  5|_|   |_|   |_|   |_|   |_|   |_|   |_|   |_|   |_|  |_|_|
|     |     |     |     |     |     |     |     |     |      |
|_____|_____|_____|_____|_____|_____|_____|_____|_____|______|

[0]
 ____________________________________________________________
|__1__|__2__|__3__|__4__|__5__|__6__|__7__|__8__|__9__|__10__|
|  -|_|   |_|   |_|   |_|   |_|   |_|   |_|   |_|   |_|  |_|_|
|     |     |     |     |     |     |     |     |     |      |
|_____|_____|_____|_____|_____|_____|_____|_____|_____|______|

[1, 2]
 ____________________________________________________________
|__1__|__2__|__3__|__4__|__5__|__6__|__7__|__8__|__9__|__10__|
|  1|2|   |_|   |_|   |_|   |_|   |_|   |_|   |_|   |_|  |_|_|
|   3 |     |     |     |     |     |     |     |     |      |
|_____|_____|_____|_____|_____|_____|_____|_____|_____|______|

[3, 4, 5, 1, 7, 0]
 ____________________________________________________________
|__1__|__2__|__3__|__4__|__5__|__6__|__7__|__8__|__9__|__10__|
|  3|4|  5|1|  7|-|   |_|   |_|   |_|   |_|   |_|   |_|  |_|_|
|   7 |  13 |  20 |     |     |     |     |     |     |      |
|_____|_____|_____|_____|_____|_____|_____|_____|_____|______|

[5, 5]
 ____________________________________________________________
|__1__|__2__|__3__|__4__|__5__|__6__|__7__|__8__|__9__|__10__|
|  5|/|   |_|   |_|   |_|   |_|   |_|   |_|   |_|   |_|  |_|_|
|     |     |     |     |     |     |     |     |     |      |
|_____|_____|_____|_____|_____|_____|_____|_____|_____|______|

[5, 5, 5]
 ____________________________________________________________
|__1__|__2__|__3__|__4__|__5__|__6__|__7__|__8__|__9__|__10__|
|  5|/|  5|_|   |_|   |_|   |_|   |_|   |_|   |_|   |_|  |_|_|
|  15 |     |     |     |     |     |     |     |     |      |
|_____|_____|_____|_____|_____|_____|_____|_____|_____|______|

[5, 5, 3, 7, 4]
 ____________________________________________________________
|__1__|__2__|__3__|__4__|__5__|__6__|__7__|__8__|__9__|__10__|
|  5|/|  3|/|  4|_|   |_|   |_|   |_|   |_|   |_|   |_|  |_|_|
|  13 |  27 |     |     |     |     |     |     |     |      |
|_____|_____|_____|_____|_____|_____|_____|_____|_____|______|

[10]
 ____________________________________________________________
|__1__|__2__|__3__|__4__|__5__|__6__|__7__|__8__|__9__|__10__|
|   |X|   |_|   |_|   |_|   |_|   |_|   |_|   |_|   |_|  |_|_|
|     |     |     |     |     |     |     |     |     |      |
|_____|_____|_____|_____|_____|_____|_____|_____|_____|______|

[10, 2]
 ____________________________________________________________
|__1__|__2__|__3__|__4__|__5__|__6__|__7__|__8__|__9__|__10__|
|   |X|  2|_|   |_|   |_|   |_|   |_|   |_|   |_|   |_|  |_|_|
|     |     |     |     |     |     |     |     |     |      |
|_____|_____|_____|_____|_____|_____|_____|_____|_____|______|

[10, 2, 4]
 ____________________________________________________________
|__1__|__2__|__3__|__4__|__5__|__6__|__7__|__8__|__9__|__10__|
|   |X|  2|4|   |_|   |_|   |_|   |_|   |_|   |_|   |_|  |_|_|
|  16 |  22 |     |     |     |     |     |     |     |      |
|_____|_____|_____|_____|_____|_____|_____|_____|_____|______|

[10, 10, 10]
 ____________________________________________________________
|__1__|__2__|__3__|__4__|__5__|__6__|__7__|__8__|__9__|__10__|
|   |X|   |X|   |X|   |_|   |_|   |_|   |_|   |_|   |_|  |_|_|
|  30 |     |     |     |     |     |     |     |     |      |
|_____|_____|_____|_____|_____|_____|_____|_____|_____|______|

[10, 10, 10, 10]
 ____________________________________________________________
|__1__|__2__|__3__|__4__|__5__|__6__|__7__|__8__|__9__|__10__|
|   |X|   |X|   |X|   |X|   |_|   |_|   |_|   |_|   |_|  |_|_|
|  30 |  60 |     |     |     |     |     |     |     |      |
|_____|_____|_____|_____|_____|_____|_____|_____|_____|______|

[3, 7, 10, 10, 5, 5]
 ____________________________________________________________
|__1__|__2__|__3__|__4__|__5__|__6__|__7__|__8__|__9__|__10__|
|  3|/|   |X|   |X|  5|/|   |_|   |_|   |_|   |_|   |_|  |_|_|
|  20 |  45 |  65 |     |     |     |     |     |     |      |
|_____|_____|_____|_____|_____|_____|_____|_____|_____|______|

[1, 2, 3, 7, 10, 5, 5, 7, 0, 0, 0, 0, 0, 0, 0, 10]
 ____________________________________________________________
|__1__|__2__|__3__|__4__|__5__|__6__|__7__|__8__|__9__|__10__|
|  1|2|  3|/|   |X|  5|/|  7|-|  -|-|  -|-|  -|-|   |X|  |_|_|
|   3 |  23 |  43 |  60 |  67 |  67 |  67 |  67 |     |      |
|_____|_____|_____|_____|_____|_____|_____|_____|_____|______|

[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 3]
 ____________________________________________________________
|__1__|__2__|__3__|__4__|__5__|__6__|__7__|__8__|__9__|__10__|
|  -|-|  -|-|  -|-|  -|-|  -|-|  -|-|  -|-|  -|-|  -|-| 3|_|_|
|   0 |   0 |   0 |   0 |   0 |   0 |   0 |   0 |   0 |      |
|_____|_____|_____|_____|_____|_____|_____|_____|_____|______|

[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4, 3, 2]
 ____________________________________________________________
|__1__|__2__|__3__|__4__|__5__|__6__|__7__|__8__|__9__|__10__|
|  -|-|  -|-|  -|-|  -|-|  -|-|  -|-|  -|-|  -|-|  -|4| 3|2|_|
|   0 |   0 |   0 |   0 |   0 |   0 |   0 |   0 |   4 |  9   |
|_____|_____|_____|_____|_____|_____|_____|_____|_____|______|

[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 5, 3, 7]
 ____________________________________________________________
|__1__|__2__|__3__|__4__|__5__|__6__|__7__|__8__|__9__|__10__|
|  -|-|  -|-|  -|-|  -|-|  -|-|  -|-|  -|-|  -|-|  -|5| 3|/|_|
|   0 |   0 |   0 |   0 |   0 |   0 |   0 |   0 |   5 |      |
|_____|_____|_____|_____|_____|_____|_____|_____|_____|______|

[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 5, 3, 7, 4]
 ____________________________________________________________
|__1__|__2__|__3__|__4__|__5__|__6__|__7__|__8__|__9__|__10__|
|  -|-|  -|-|  -|-|  -|-|  -|-|  -|-|  -|-|  -|-|  -|5| 3|/|4|
|   0 |   0 |   0 |   0 |   0 |   0 |   0 |   0 |   5 | 19   |
|_____|_____|_____|_____|_____|_____|_____|_____|_____|______|

[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
 ____________________________________________________________
|__1__|__2__|__3__|__4__|__5__|__6__|__7__|__8__|__9__|__10__|
|  -|-|  -|-|  -|-|  -|-|  -|-|  -|-|  -|-|  -|-|  -|-| -|-|_|
|   0 |   0 |   0 |   0 |   0 |   0 |   0 |   0 |   0 |  0   |
|_____|_____|_____|_____|_____|_____|_____|_____|_____|______|

